### ISSUE TITLE
Batched tf.linalg.logdet produces inconsistent NaN mask on 2x2 float32 matrices vs PyTorch

### ISSUE THREAD

**Issue by tingPetty** [tensorflow/tensorflow#111469]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.19

### Custom code

Yes

### OS platform and distribution

Windows 11

### Python version

3.10.18

### Current behavior?

When executing `tf.linalg.logdet` on batched 2x2 float32 tensors (shape `[4, 4, 2, 2]`), the function resolves almost the entire batch to `NaN`, heavily mismatching PyTorch's output. While both frameworks properly fail on the 9 internal matrices that possess non-positive determinants, TensorFlow additionally outputs `NaN` for matrices that PyTorch evaluates mathematically without issue.

This happens because TensorFlow imposes strict symmetric positive-definite constraints on the entire batch. If matrices within the batch are merely non-symmetric (yet possess valid positive determinants), TF's batched Cholesky solver fails and outputs `NaN`.

Expected behavior: For matrices with positive determinants, `tf.linalg.logdet` should successfully compute the scalar values akin to PyTorch, rather than populating the output tensor with unpredictable `NaN` evaluations.

### Standalone code to reproduce the issue

```shell
import numpy as np
import torch
import tensorflow as tf

np.random.seed(42)
input_np = np.random.randn(4, 4, 2, 2).astype(np.float32)

input_pt = torch.tensor(input_np)
out_pt = torch.logdet(input_pt)

input_tf = tf.constant(input_np)
out_tf = tf.linalg.logdet(input_tf)

pt_np_out = out_pt.numpy()
tf_np_out = out_tf.numpy()
max_diff = np.max(np.abs(pt_np_out - tf_np_out))
neg_det_count = np.sum(np.linalg.det(input_np) <= 0)
print(f"Number of matrices with non-positive determinant: {neg_det_count}")
print(f"PyTorch logdet output: {pt_np_out}")
print(f"TensorFlow logdet output: {tf_np_out}")
print(f"Maximum difference: {max_diff}")
```

### Relevant log output

```shell
Number of matrices with non-positive determinant: 9
PyTorch logdet output: [[-0.16716182 -1.6604435  -0.7548522          nan]
 [ 0.5398615          nan         nan         nan]
 [-0.12048236         nan         nan         nan]
 [-0.8231379          nan         nan -1.0002422 ]]
TensorFlow logdet output: [[-1.0876427        nan        nan        nan]
 [       nan        nan        nan        nan]
 [       nan        nan        nan        nan]
 [       nan        nan        nan        nan]]
Maximum difference: nan
```

**Comment by avmohansai** [comment]
Hi [@Venkat6871](https://github.com/Venkat6871) ,
I Was able to reproduce the issue on Colab using `TF v2.19.0`,TF` v2.20.0` and `TF-nightly` ,Please find the [gist](https://colab.sandbox.google.com/gist/avmohansai/867b944887916f209b14f051c0729a82/111469_tf_2-20-0_nightly.ipynb) here for reference.
